### PR TITLE
Add tools directory, md2swagger

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

File: tools/md2swagger.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        properties['format']='date-time'
+    elif type=='double':
+        properties['type']='number'
+        properties['format']='double'
+    elif type=='integer':
+        properties['type']='integer'
+        properties['format']='int64'
+    elif type=='string':
+        properties['type']='string'
+    return properties;
```
Comment (dcrissman) [inline tools/md2swagger.py]: how about a default case if unexpected type is passed in

File: tools/md2swagger.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+def singular(s):
+    if s[-3:]=='ies':
+        return s[0:-3]+'y'
+    elif s[-4:]=='sses':
+        return s[0:-4]+'ss'
+    elif s[-1]=='s':
+        return s[0:-1]
+    else:
+        return s;
```
Comment (dcrissman) [inline tools/md2swagger.py]: Seems frail

**Comment by bserdar** [comment]
I'll add the default case for unknown type (string).

Do you have a better idea to convert a plural to a singular?

File: tools/md2swagger.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                fieldBlock['items']={'$ref':'#definitions/'+newEntityName}
+                toplevel[newEntityName]=processFieldTree(fieldContents['fields'],toplevel,nameStack)
+                nameStack.pop()
+            elif fieldContents['type']=='array':
+                fieldBlock['type']='array'
+                arrayType=fieldContents['items']['type']
+                if arrayType=='object':
+                    nameStack.append(singular(capital1(field)))
+                    newEntityName=makeEntityName(nameStack)
+                    fieldBlock['items']={'$ref':'#definitions/'+newEntityName}
```
Comment (alechenninger) [inline tools/md2swagger.py]: I made some fixes in the internal repo for this you may want to grab
- need #/definitions (missing /)
- The new top level sub entities need type: object and properties: fields

File: tools/md2swagger.py (hunk 2 above)
Comment (dcrissman) [inline tools/md2swagger.py]: > Do you have a better idea to convert a plural to a singular?

In fairness, no. My thought was that a word that correctly ends with 's' would have problems, but the more I think about it, that may be a fringe case anyway.

Could you explain the need for this conversion?

**Comment by bserdar** [comment]
Without this, we had UserLegalEntities as a classname.

File: tools/md2swagger.py (hunk 2 above)
Comment (dcrissman) [inline tools/md2swagger.py]: > Without this, we had UserLegalEntities as a classname.

I won't hold this up further on this point. This is certainly good enough and we can adjust if/when we find a use case. Thanks.

**Comment by bserdar** [comment]
Added fixes from the internal repo, and added a catch-all for type
properties.

**Pull request merged**